### DISCUSSION TITLE
Why the generated "GetHashCode" uses this number ?

### DISCUSSION THREAD

**Discussion by a510** [dotnet/roslyn#65251] [Q&A] [answered]
The generated `GetHashCode` [here](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQskUAJgAQBKAFAJYB2ALuQgJQDcQA===) multiplies the hash code of each field by "-1521134295".
1. why is this necessary?
2. why not just pass the fields to "HashCode.Combine" ?
3. should I do this in my code as well ?
Thanks

**Comment by CyrusNajmabadi** [comment, marked as answer]
It's just a nice large prime number (when unsigned). This code emit existed long before System.Hashcode was around. And it has no dependency on it. You do not need to do this.
